### DISCUSSION TITLE
How do I format my xcassets file to get the namespacing shown in the readme?

### DISCUSSION THREAD

**Discussion by oliverfoggin** [SwiftGen/SwiftGen#945] [General]
I've tried multiple ways to set up the assets catalog and different config options in the yml but I can't get the output to look like it does in the readme with multiple namespaces...

![image](https://user-images.githubusercontent.com/1813244/173334421-e863720f-6571-4bd8-bab4-1c6dc8d3262c.png)

All I get is a single flat enum with all the images in it...

```
public enum Asset {
  public static let arrowBack = ImageAsset(name: "arrow-back")
  public static let arrowBottomRight = ImageAsset(name: "arrow-bottom-right")
  public static let arrowClockwise = ImageAsset(name: "arrow-clockwise")
  public static let arrowDouble = ImageAsset(name: "arrow-double")
  public static let arrowDown = ImageAsset(name: "arrow-down")
  public static let arrowTopRight = ImageAsset(name: "arrow-top-right")
```

etc...

How can I create the nested enums like they are shown in the readme.

Thansk

**Comment by djbe** [comment]
You have to check the `provides namespace` toggle for groups in your asset catalog.
Random link from the internet:
![Sceenshot](https://i.stack.imgur.com/GKQNe.png)


Or you can achieve this without checking that toggle, by setting the `forceProvidesNamespaces` parameter in your SwiftGen config, [see the docs](https://github.com/SwiftGen/SwiftGen/blob/stable/Documentation/templates/xcassets/swift5.md#customization).

**Reply by oliverfoggin** [reply]
Hmm... I did try that but it didn't seem to do anything. I've also tried using the `forceProvidesNamespaces` but that didn't seem to work either.

I think it might be the assets file. I'll try again.

Thanks 😄

**Comment by oliverfoggin** [comment]
@djbe ok... so I tried this earlier and it didn't work. But now it's working. 😅 

I must have made a typo or something before.

Thanks for the help.
